src/main/java/dk/kb/util/Resolver.java

```diff
@@ -140,21 +142,55 @@ public static URL resolveURL(String resourceName)
             // Overall principle is recursive descend, but only visiting the folders that are viable candidates.
             // This is done by splitting the glob in segments, one segment per folder, then walking down the hierarchy
             // while ensuring that each step in the hierarchy matches the right segment of the glob.
-            String absoluteGlob = prefix + glob;
-            Stream<String> segments = Arrays.stream(absoluteGlob.split(Pattern.quote(File.separator)));
-            List<PathMatcher> matchers = segments.
-                    filter(segment -> !segment.isEmpty()). // The first segment is always empty due to the splitting
+            List<PathMatcher> matchers =
+                    extractSegments(prefix + glob).stream().
                     map(segment -> FileSystems.getDefault().getPathMatcher("glob:" + segment)).
                     collect(Collectors.toList());
             // This simply resolves to {@code /} on unix systems, but there can be multiple roots on Windows
             for (File root: File.listRoots()) {
                 walkMatches(root.toPath(), matchers, 0, paths::add);
             }
         }
-        Collections.sort(paths);
-        log.debug("Resolved glob '" + glob + "' to " + paths);
-        return paths;
+
+        
+        List<Path> result = new ArrayList<>(paths);
+        Collections.sort(result);
+        log.debug("Resolved glob '" + glob + "' to " + result);
+        return result;
     }
+
+    /**
+     * Takes an absolute glob (a glob starting at the root).
+     * Splits on {@link File#separator}, removes empty segments and {@code .}, collapses {@code ..}.
+     * @param glob a file system glob expression (plain paths are valid globs).
+     * @return the segments or empty if reduction of {@code ..} leads to a non-viable situation.
+     */
+    static List<String> extractSegments(String glob) {
+        // Handle empty and .
+        List<String> segments = Arrays.stream(glob.split(Pattern.quote(File.separator))).
+                filter(segment -> !segment.isEmpty() && ! segment.equals(".")).
+                collect(Collectors.toList());
+        if (segments.isEmpty()) {
+            return segments;
+        }
+
+        // Handle ..
+        int pos = 0;
+        while (pos < segments.size()-1) {
+            if (pos == 0 && "..".equals(segments.get(0))) { // Going further up than the root does not make sense
+                return Collections.emptyList();
+            }
+            if ("..".equals(segments.get(pos+1))) { // The subsequent segment goes one level up
+                segments.remove(pos+1);
+                segments.remove(pos);
+                pos = Math.max(0, pos-1); // Go one segment level back and try again
+                continue;
```

Comment:
Given that the glob: "../foo/bar/baz/*.yaml" would yield a empty list, should the glob: "/foo/../../bar/baz/*.yaml" get the same treatment?

Reply:
The glob `../foo/bar/baz/.yaml` is not an "absolute glob". But if you mean `/../foo/bar/baz/.yaml` vs. `/foo/../..//foo/bar/baz/.yaml` then yes.

Reply:
Aaand I can see that I do not handle that properly in the code. I'll add a unit test and fix it!

Reply:
Sorry, meant "/../foo/bar/baz/*.yaml".

Reply:
Done.